src/libs/blueprint/conduit_blueprint_o2mrelation.hpp

```diff
@@ -96,11 +96,22 @@ bool CONDUIT_BLUEPRINT_API verify(const std::string &protocol,
 //-----------------------------------------------------------------------------
 
 //-----------------------------------------------------------------------------
-void CONDUIT_BLUEPRINT_API query_paths(const conduit::Node &o2mrelation,
-                                        conduit::Node &res);
+std::vector<std::string> CONDUIT_BLUEPRINT_API data_paths(const conduit::Node &o2mrelation);
 
 //-----------------------------------------------------------------------------
-void CONDUIT_BLUEPRINT_API to_compact(conduit::Node &o2mrelation);
+void CONDUIT_BLUEPRINT_API compact_to(const conduit::Node &o2mrelation,
```

Comment:
q: will `compact_to` be confused with existing `compact_to` methods, or is that a good thing (very similar concept?)

Reply:
I think the two functions are very similar in intent, so I consider the overloading to be helpful rather than hurtful in this case. Here's a very quick overview of what both functions do with simple examples:

#### `conduit::Node::compact_to(conduit::Node &dst)` ####

Given a node with an arbitrary stride/offset (the instance node), populate another node (the argument node) with a compacted copy of the first node's strided/offset data.

```
src: [-1.0, 1.0, -1.0, -1.0, 2.0, -1.0, -1.0] // DataType::float32(2, sizeof(float32), 3 * sizeof(float32))
dst: [1.0, 2.0]                               // DataType::float32(2, 0, 3 * sizeof(float32))
```

#### `conduit::Node::compact_to(const conduit::Node &src, conduit::Node &dst)` ####

Given a node with an arbitrary one-to-many layout (the first argument), populate another node (the second argument) with a compacted copy of the first node's one-to-many data.

```
src: 
  data: [1.0, -1.0, 2.0, 3.0, -1.0, 4.0, 5.0, 6.0]
  sizes: [1, 2, 3]
  offsets: [0, 2, 5]
dst:
  data: [1.0, 2.0, 3.0, 4.0, 5.0, 6.0]
  sizes: [1, 2, 3]
  offsets: [0, 1, 3]
```

Reply:
sounds good then!